### PR TITLE
Put all memory restrictions under a flag and enable it in the CI

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: benchmark/streamly-benchmarks.cabal [hunk 1]
```diff
@@ -144,6 +149,7 @@ library
 
 common bench-options
   import: compile-options, optimization-options, bench-depends
+  -- XXX The use of -retsopts is for runtime?
```
Comment (pranaysashank) [inline benchmark/streamly-benchmarks.cabal:152]: yes?

File: benchmark/streamly-benchmarks.cabal [hunk 2]
```diff
@@ -19,6 +19,11 @@ flag fusion-plugin
   manual: True
   default: False
 
+flag limit-memory
```
Comment (pranaysashank) [inline benchmark/streamly-benchmarks.cabal:22]: huh one more flag, but its in benchmarks so its fine maybe?

**Review by pranaysashank — APPROVED** [review]

File: benchmark/streamly-benchmarks.cabal (hunk 1 above)
Comment (adithyaov) [reply benchmark/streamly-benchmarks.cabal:152]: Cool

Comment (adithyaov) [reply benchmark/streamly-benchmarks.cabal:152]: Just to be clear about the question,
```
ghc-options: +RTS ... -RTS
```
Limits the memory for ghc itself while compilation.
Whereas,
```
ghc-options: -rtsopts +RTS ... -RTS
```
is for limiting the memory when the executable is running.
@pranaysashank Is my understanding correct?

File: .circleci/config.yml [hunk 3]
```diff
@@ -185,7 +186,8 @@ jobs:
            environment:
              BUILD: "cabal-v2"
              GHCVER: "8.4.4"
-             CABAL_BUILD_OPTIONS: "--flag debug"
+             # XXX This hopefully overrides the <<*env ?
```
Comment (adithyaov) [inline .circleci/config.yml:189]: @pranaysashank Does this override?

File: benchmark/streamly-benchmarks.cabal (hunk 2 above)
Comment (harendra-kumar) [inline benchmark/streamly-benchmarks.cabal:22]: How about naming it `limit-build-mem` to convey that it is build time?

**Pull request merged**